### PR TITLE
[VitisIP] Fixes for archgenlib

### PR DESCRIPTION
**Pull request by Ralender**

### CODE REVIEW

File: clang/lib/Frontend/InitPreprocessor.cpp [hunk 1]
```diff
@@ -1294,6 +1294,10 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
     Builder.defineMacro(
         "__VITIS_KERNEL",
         "extern \"C\" __attribute__((annotate(\"vitis_kernel\")))");
+  else
+    /// This is needed to compile ArchGenLib but I think this else case is not
+    /// the correct way to address the issue
```
Comment (keryell) [inline clang/lib/Frontend/InitPreprocessor.cpp]: Why we have this else case, by the way?

**Review by keryell — COMMENTED** [review]
It is not clear what is done here and why.

File: clang/lib/Frontend/InitPreprocessor.cpp (hunk 1 above)
Comment (Ralender) [reply clang/lib/Frontend/InitPreprocessor.cpp]: We experimented with this while using the VitisIP flow on archgen lib. and it did remove some preprocessor in some simple cases. but it prevents from detecting what is the compilation mode. which is a lot worse.

**Review by keryell — APPROVED** [review]
Thanks!

**Pull request merged**